cmd/main.go

```diff
@@ -468,8 +475,10 @@ func main() {
 	// initialize things
 	profile := vault.Profile{
 		Name: options.AwsProfile,
-		RoleARN: fmt.Sprintf("arn:aws:iam::%v:role/%v",
-			options.AwsAccountID, options.Role),
+		RoleARN: fmt.Sprintf("arn:%s:iam::%d:role/%s",
+			getPartition(options.AwsRegion),
```

Comment:
```suggestion
			partition,
```
Then add above something to call `getPartition` like:
```
partition, err := getPartition(options.AwsRegion)
	if err != nil {
		fmt.Println(err)
		fmt.Println("Using aws as partition")
		partition = "aws"
	}
```
If you want to avoid noping out of the process